ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -12016,6 +12016,10 @@ void HqlCppTranslator::buildCppFunctionDefinition(BuildCtx & ctx, IHqlExpression
     }
 
     MemberFunction userFunc(*this, ctx, proto, MFdynamicproto);
+    userFunc.ctx.addQuoted("#undef EmbedCodeBlock");
+    VStringBuffer eblock("EmbedCodeBlock __b%d;", rand());
```

Comment:
You can't use rand() - you need to use some consistent numbering scheme.

Reply:
I suspect you have done it so that the generated code cannot know the name, but it would make it impossible to check the regression suite output.

Reply:
That did occur to me, but I'm sure there's a way of making it consistent when running regression suite without the name being known. I didn't bother fixing that issue for this POC though. As you surmise, it's to make it harder for embedded code to explicitly call destructor.